src/PFire.Core/Protocol/Messages/Bidirectional/ChatMessage.cs

```diff
@@ -52,7 +52,8 @@ public override async Task Process(IXFireClient context)
                     break;
                 
                 case ChatMessageType.Acknowledgement:
-                    // Do nothing for now.
+                    var ack = BuildChatMessageResponse(context.SessionId);
```

Comment:
If I'm understanding, this is the flow we're achieving? This seems unintuitive to me, but I'm hyped as hell if this works.
`Sender -> ContentMessage -> Server -> ContentMessage -> Recipient -> AcknowledgementMessage -> Server -> ContentMessage -> Sender`

Reply:
From all the protocol API I read, this is the flow. The original client is waiting for an acknowledgement of the message in a timer.